invisible_cities/io/dst_io.py

```diff
@@ -29,38 +29,53 @@ def load_dsts(dst_list, group, node):
     dsts = [load_dst(filename, group, node) for filename in dst_list]
     return pd.concat(dsts, ignore_index=True)
 
-def _make_tabledef(column_types : pd.Series, str_col_length : int=32) -> dict:
+def _make_tabledef(column_types : np.dtype, str_col_length) -> dict:
     tabledef = {}
-    for indx, colname in enumerate(column_types.index):
+    for indx, colname in enumerate(column_types.names):
         coltype = column_types[colname].name
         if coltype == 'object':
             tabledef[colname] = tb.StringCol(str_col_length, pos=indx)
         else:
             tabledef[colname] = tb.Col.from_type(coltype, pos=indx)
     return tabledef
 
-def _store_pandas_as_tables(h5out : tb.file.File, df : pd.DataFrame, group_name : str, table_name : str, compression : str='ZLIB4', descriptive_string : [str]="", str_col_length : int=32) -> None:
-    if len(df) == 0:
-        warnings.warn(f'dataframe is empty', UserWarning)
+def _can_cast(arr : np.ndarray, table_types : np.dtype):
+    arr_types = arr.dtype
+    if set(arr_types.names) != set(table_types.names):
+        raise TableMismatch(f'dataframe differs from already existing table structure')
+    for name in arr_types.names:
+        if arr_types[name].name == 'object':
+            max_str_length = max(map(len, arr[name]))
+            if max_str_length > table_types[name].itemsize:
+                warnings.warn(f'dataframe contains strings longer than allowed', UserWarning)
+        elif arr_types[name] != table_types[name]:
+            warnings.warn(f'dataframe numeric types not consistent with the table existing ones', UserWarning)
+
+
+def store_pandas_as_tables(h5out : tb.file.File, df : pd.DataFrame, group_name : str, table_name : str, compression : str='ZLIB4', descriptive_string : [str]="", str_col_length : int=32) -> None:
```

Comment:
I know it was already like that, but can we take the opportunity to split this into many for readability?